### DISCUSSION TITLE
YOURLs install is not pointing the short urls the unshortened main url

### DISCUSSION THREAD

**Discussion by phillycodehound** [YOURLS/YOURLS#3800] [Q&A]
Hi

YOURLs install is not pointing the short urls the unshortened main url on SOCL.bz

Example: https://socl.bz/mbertoni 

should redirect to: 

https://entrepreneursenigma.com/podcast/michael-bertoni-on-saas-recruiting-and-the-entrepreneurial-experience/

But it's not.

It seems as though the 301 redirect isn't connecting. How do I fix this?

Thanks!

**Comment by dgw** [comment]
Please double check your [server configuration](https://yourls.org/docs/guide/server-configuration) and follow the [troubleshooting steps](https://yourls.org/docs/category/troubleshooting).

**Reply by phillycodehound** [reply]
Ran through both. Everything seems right. But still just getting Hello World!

Also recreated the DB again. Now I'm getting a 404 not found on all the short URLS.

I migrated the DB tables over from an (same version) mySQL. Getting rid of that host.

**Reply by dgw** [reply]
And yet, it's definitely a server configuration issue, because YOURLS doesn't have this Hello World messaging.

Without some more details about your environment and config, no one will be able to help you.

**Reply by phillycodehound** [reply]
Well that was a rogue index file.

I'm on a VPS nginx running Ubuntu off of Hostinger.

I moved the tables

yourls_log, yourls_options, yourls_url over from the other host db. In the past this has worked to migrate the data. This time the connections to the main URL are broken.

**Reply by dgw** [reply]
[404 instead of redirection is one of the most common problems](https://yourls.org/docs/guide/troubleshooting/frequent-issues#404-instead-of-redirection), and it is related to the nginx (or Apache, for installs using that) configuration. I can't say any more than that without seeing the config.

**Reply by phillycodehound** [reply]
I can share the config.

**Reply by dgw** [reply]
Server config...

**Reply by dgw** [reply]
And you should now change your cookie key, perhaps also the database password.

**Reply by phillycodehound** [reply]
Man, I was hopeful! But still nope!

**Reply by dgw** [reply]
You only needed to change those because they were in the YOURLS config shared before.

The _server_ config (nginx) is needed to debug 404s where shorturl redirections should be.

**Reply by phillycodehound** [reply]
This is what Debug says: 


Connected to database soclbz on localhost 
SQL perform: SELECT option_name, option_value FROM yourls_options WHERE 1=1 (0.00046 s)
loaded /home/seth/htdocs/socl.bz/user/plugins/hyphens-in-urls/plugin.php
Check for new version: no
SQL perform: SELECT COUNT(keyword) as count, SUM(clicks) as sum FROM `yourls_url` WHERE 1=1  (0.00173 s)
SQL perform: SELECT * FROM `yourls_url` WHERE 1=1  ORDER BY `timestamp` desc LIMIT 0, 15; (0.00033 s)

**Reply by phillycodehound** [reply]
and does this help?

```nginx
server {
  listen 80;
  listen [::]:80;
  listen 443 ssl http2;
  listen [::]:443 ssl http2;
  {{ssl_certificate_key}}
  {{ssl_certificate}}
  server_name www.socl.bz;
  return 301 https://socl.bz$request_uri;
}

server {
  listen 80;
  listen [::]:80;
  listen 443 ssl http2;
  listen [::]:443 ssl http2;
  {{ssl_certificate_key}}
  {{ssl_certificate}}
  server_name socl.bz www1.socl.bz;
  {{root}}

  {{nginx_access_log}}
  {{nginx_error_log}}

  if ($scheme != "https") {
    rewrite ^ https://$host$uri permanent;
  }

  location ~ /.well-known {
    auth_basic off;
    allow all;
  }

  {{settings}}

  location / {
    {{varnish_proxy_pass}}
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-Host $http_host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_hide_header X-Varnish;
    proxy_redirect off;
    proxy_max_temp_file_size 0;
    proxy_connect_timeout      720;
    proxy_send_timeout         720;
    proxy_read_timeout         720;
    proxy_buffer_size          128k;
    proxy_buffers              4 256k;
    proxy_busy_buffers_size    256k;
    proxy_temp_file_write_size 256k;
  }

  location ~* ^.+\.(css|js|jpg|jpeg|gif|png|ico|gz|svg|svgz|ttf|otf|woff|woff2|eot|mp4|ogg|ogv|webm|webp|zip|swf|map|mjs)$ {
    add_header Access-Control-Allow-Origin "*";
    expires max;
    access_log off;
  }

  location ~ /\.(ht|svn|git) {
    deny all;
  }

  if (-f $request_filename) {
    break;
  }
}

server {
  listen 8080;
  listen [::]:8080;
  server_name socl.bz www1.socl.bz;
  {{root}}

  try_files $uri $uri/ /index.php?$args;
  index index.php index.html;

  location ~ \.php$ {
    include fastcgi_params;
    fastcgi_intercept_errors on;
    fastcgi_index index.php;
    fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
    try_files $uri =404;
    fastcgi_read_timeout 3600;
    fastcgi_send_timeout 3600;
    fastcgi_param HTTPS "on";
    fastcgi_param SERVER_PORT 443;
    fastcgi_pass 127.0.0.1:{{php_fpm_port}};
    fastcgi_param PHP_VALUE "{{php_settings}}";
  }

  if (-f $request_filename) {
    break;
  }
}
```

**Reply by phillycodehound** [reply]
Here's a test account:

https://socl.bz/admin

pwd taken out for security

**Reply by dgw** [reply]
If the port 80/443 listener is what's handling YOURLS, it is missing the `try_files` piece and some other things from the example configuration.

You say you have run through the server configuration and troubleshooting pages, but what you have is so complicated it'll be hard to debug. (For example, what's Varnish doing in there? It's handling requests under `/` that I'd expect for YOURLS to handle instead, and the port 8080 listener that looks configured mostly correctly is unreachable from where I am.)

**Reply by phillycodehound** [reply]
I will investigate further with the support at Hostinger and will post here what they say to help others.
